src/gc.c

```diff
@@ -344,6 +345,16 @@ static void run_finalizers(void)
     JL_GC_POP();
 }
 
+void jl_gc_inhibit_finalizers(int state)
+{
+    if (jl_gc_finalizers_inhibited && !state && !jl_in_gc) {
+        jl_in_gc = 1;
```

Comment:
I think it was a concern that finalizers could need to be codegenned which might trigger gc again